### ISSUE TITLE
Native Wayland support in flatpak builds

### ISSUE THREAD

**Issue by MarkGhanz** [yuzu-emu/yuzu#10792]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

I think it's good to have native wayland support in flatpak builds. The current latest flatpak build 1469 needs xwayland to launch.

### Why would this feature be useful?

Since yuzu is in fast development, it's hard for package maintainers in distros to keep up, but native wayland support seems working on native builds instead of flatpak ones.

**Comment by liamwhite** [comment]
It is hard to recommend enabling Wayland on officially-sanctioned distributions due to two issues:
- OpenGL requires using EGL unconditionally on Wayland. This absolutely destroys performance because it forces the use of a single-threaded context.
- Popular compositors cap the Vulkan swapchain to the refresh rate of the monitor being displayed on, meaning unlocked framerate doesn't work correctly.

**Comment by GioPanaro** [comment]
@liamwhite true, but wayland is the future of linux whether we like it or not - or at the very least, x11 is not the future. Xorg is EOL and currently has bugs that will more than likely never be fixed. Wayland is evolving and growing, even though it feels very unstable at points.
